.github/workflows/build-notebooks-pr.yaml

```diff
@@ -1,10 +1,13 @@
 ---
 "name": "Build Notebooks"
-"permissions":
-  "packages": "read"
 "on":
   "pull_request":
 
+permissions:
+  contents: read
+  packages: read
```

Comment:
I see we don't interact with GitHub Packages in this repository, so maybe the `packages` permission is not necessary here. I know it was there before, but just pointing it out.

Reply:
`packages: read` is necessary to use ghcr.io repository as a cache; it is running podman with

```
--cache-from ghcr.io/opendatahub-io/notebooks/workbench-images/build-cache
```

Reply:
looking at the trial PR, it does not seem to be pulling from cache, not even the base images which should be completely unchanged! https://github.com/opendatahub-io/notebooks/actions/runs/9498328161/job/26176821937?pr=555

In the recent PR from Diamond it is using the cache just fine; not sure what busted the cache here https://github.com/opendatahub-io/notebooks/actions/runs/9490922279/job/26155372847?pr=557

Reply:
it's not pulling from cache because the `FROM registry.access.redhat.com/ubi9/python-39:latest` image has been updated in the meantime! that explains things, mystery solved, never mind me